src/core/actions/unitactions/Move.java

```diff
@@ -1,45 +1,170 @@
 package core.actions.unitactions;
 
+import core.TechnologyTree;
+import core.Types;
 import core.actions.Action;
+import core.game.Board;
 import core.game.GameState;
 import core.actors.units.Unit;
+import utils.Vector2d;
+import utils.graph.NeighbourProvider;
+import utils.graph.TreeNode;
+import utils.graph.TreePathfinder;
 
+import java.util.ArrayList;
 import java.util.LinkedList;
+import java.util.Vector;
 
 public class Move extends UnitAction
 {
-    private int destX;
-    private int destY;
+    private Vector2d destination;
 
     public Move(Unit u)
     {
         super.unit = u;
     }
 
-    public void setDest(int x, int y) {this.destX = x; this.destY = y;}
-    public int getDestX() {
-        return destX;
-    }
-    public int getDestY() {
-        return destY;
-    }
+    public void setDestination(Vector2d destination) {this.destination = destination; }
+    public Vector2d getDestination() { return destination; }
 
     @Override
-    public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO: compute all the possible Move actions for super.unit.
-        return new LinkedList<>();
+    public LinkedList<Action> computeActionVariants(final GameState gs)
+    {
+        LinkedList<Action> moves = new LinkedList<>();
+        TreePathfinder tp = new TreePathfinder(unit.getPosition(), new StepMove(gs, unit));
+
+        //If a units turn is FINISHED don't do unnecessary calculations.
+        if(unit.checkStatus(Types.TURN_STATUS.MOVED)) {
+            for(TreeNode tile : tp.findPaths()) {
+                Move action = new Move(unit);
+                action.setDestination(tile.getPosition());
+
+                if(action.isFeasible(gs)) {
```

Comment:
We don't need to call isFeasible, we know it already is because tp.findPaths() only retrieves feasible paths. It's using the StepMove class below, which considers obstacles and move range already.